### PR TITLE
Visual updates to expander

### PR DESCRIPTION
**Pull request by tashatitova**
The expander control was pretty basic so added styles to it.

- Removed the extra xaml page (Expander) and moved all styles to themeresources
- Added system color brushes for high contrast as only two of them were available, and using the extra levels of abstraction through existing SystemControl brushes made no sense for new control
- Simplified the chevron logic via adding a tag to style but this will be revisited with AnimatedIcon chevron in the future
- Various brushes, corner radius, layout, margins, etc updates

Future work: - Pop out animations updates

Screenshots:

Light: regular/hover/press

Light
Light-hover
Light-press

Dark: regular/hover/press

Dark
Dark-hover
Dark-press

NOTE: for high contrast, I'm using custom ButtonText and ButtonFace colors for the header/toggle, to make sure they are differentiating from WindowText + WindowColor for the drop down

Regular/hover + press (these two look the same)

HC
HC-hover-press

### CODE REVIEW

File: dev/Expander/Expander.vcxitems [hunk 1]
```diff
@@ -24,11 +24,6 @@
     <ClCompile Include="$(MSBuildThisFileDirectory)ExpanderAutomationPeer.cpp" />
   </ItemGroup>
   <ItemGroup>
-    <Page Include="$(MSBuildThisFileDirectory)Expander.xaml">
```
Comment (karenbtlai) [inline dev/Expander/Expander.vcxitems:27]: The convention is to have the template in the `Expander.xaml`, with resources such as brushes in `Expander_themeresources.xaml`. I would revert this change.

File: dev/Expander/Expander.vcxitems [hunk 2]
```diff
@@ -24,11 +24,6 @@
...
-      <UseVisualStyle>Any</UseVisualStyle>
-      <Version>RS1</Version>
-      <Type>DefaultStyle</Type>
-    </Page>
```
Comment (karenbtlai) [inline dev/Expander/Expander.vcxitems:31]: The convention is to have the template in the `Expander.xaml`, with resources such as brushes in `Expander_themeresources.xaml`. I would revert this change.

**Review by karenbtlai — CHANGES_REQUESTED** [review]

File: dev/Expander/Expander.vcxitems (hunk 2 above)
Comment (tashatitova) [reply dev/Expander/Expander.vcxitems:31]: I was going with previous controls I worked on - they all had just one themeresources file (ContentDialog, Menuflyout etc) including the template itself. Having two files only splits files into two places so you have to search separately when updating styles. @teaP @StephenLPeters can you comment on this please?

**Review by marksfoster — APPROVED** [review]

File: dev/Expander/Expander_themeresources.xaml [hunk 3]
```diff
@@ -2,31 +2,467 @@
...
+    <x:String x:Key="ExpanderChevronUpGlyph">&#xE70E;</x:String>
+    <x:String x:Key="ExpanderChevronDownGlyph">&#xE70D;</x:String>
+    <x:Double x:Key="ExpanderChevronSize">30</x:Double>
+    <CornerRadius x:Key="ExpanderCornerRadius">2</CornerRadius>
     <x:Double x:Key="ExpanderPopinVerticalOffset">50</x:Double>
+    <x:Double x:Key="ExpanderMinHeight">64</x:Double>
+    <Thickness x:Key="ExpanderPadding">16,0,16,0</Thickness>
+    <Thickness x:Key="ExpanderHeaderPadding">16,0,0,0</Thickness>
+
+    <Style TargetType="local:Expander" BasedOn="{StaticResource DefaultExpanderStyle}"/>
```
Comment (ranjeshj) [inline dev/Expander/Expander_themeresources.xaml]: Please verify with @canhua. These styles might need to go into expander.xaml

File: Expander.xaml [hunk 4]
```diff
@@ -0,0 +1,381 @@
...
+    <Style TargetType="local:Expander" BasedOn="{StaticResource DefaultExpanderStyle}"/>
+
+    <Style x:Key="DefaultExpanderStyle" TargetType="local:Expander">
+        <!-- We won't make this control tab stoppable, since the tab focus should go
+            to the toggle button. For this logic, several things were made:
+            1. Expander::OnApplyTemplate - We set the toggle button's events source to the expander's. This will announce expander properties
+               (expand/collapse) changes to narrator. Without this, narrator would announce the toggle button's "on/off" logic
+               instead of the "expand/collapse" logic.
+            2. ExpanderAutomationPeer::GetChildrenCore - Because of 1., we need to override GetChildrenCore to NOT include the toggle button.
+               If we don't do this, we create a weird dependency cycle between the toggle button and the expander.
```
Comment (ranjeshj) [inline Expander.xaml:17]: There seem to be two Expander.xaml files. This content should be in the one under dev\Expander folder

**Comment by ranjeshj** [comment]
Please merge in master to get a fix for the failing tests.

File: Expander.xaml (hunk 4 above)
Comment (tashatitova) [reply Expander.xaml]: Yes I fixed this

**Comment by mdtauk** [comment]
What is the thinking behind the large padding above and below the heading text.  The padding around the close button visual is uneven too, which will not look good with the hover and pressed state backing plates.

File: dev/Expander/Expander_themeresources.xaml [hunk 5]
```diff
@@ -2,31 +2,464 @@
...
+    <x:String x:Key="ExpanderChevronUpGlyph">&#xE70E;</x:String>
+    <x:String x:Key="ExpanderChevronDownGlyph">&#xE70D;</x:String>
+    <x:Double x:Key="ExpanderChevronSize">30</x:Double>
+    <CornerRadius x:Key="ExpanderCornerRadius">2</CornerRadius>
     <x:Double x:Key="ExpanderPopinVerticalOffset">50</x:Double>
+    <x:Double x:Key="ExpanderMinHeight">64</x:Double>
+    <Thickness x:Key="ExpanderPadding">16,0,16,0</Thickness>
+    <Thickness x:Key="ExpanderHeaderPadding">16,0,0,0</Thickness>
+
+    <Style x:Key="DefaultExpanderStyle" TargetType="local:Expander">
```
Comment (karenbtlai) [inline dev/Expander/Expander_themeresources.xaml:95]: This style needs to be moved to Expander.xaml as it is the default style.

File: dev/Expander/Expander.xaml [hunk 6]
```diff
@@ -19,10 +20,21 @@
                Without this override, the narrator user that focuses the expander on a touch screen will see that pressing "Tab"
                doesn't work how they would expect.-->
         <Setter Property="IsTabStop" Value="False"/>
+        <Setter Property="Background" Value="{ThemeResource ExpanderBackground}" />
+        <contract7Present:Setter Property="CornerRadius" Value="{ThemeResource ControlCornerRadius}" />
```
Comment (karenbtlai) [inline dev/Expander/Expander.xaml:24]: nit: I think this should be moved to the bottom under `VerticalContentAlignment` for consistent formatting

File: dev/Expander/Expander.xaml [hunk 7]
```diff
@@ -2,10 +2,11 @@
 <ResourceDictionary
     xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
     xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
-    xmlns:local="using:Microsoft.UI.Xaml.Controls"
-    xmlns:contract7Present="http://schemas.microsoft.com/winfx/2006/xaml/presentation?IsApiContractPresent(Windows.Foundation.UniversalApiContract,7)">
+    xmlns:contract7Present="http://schemas.microsoft.com/winfx/2006/xaml/presentation?IsApiContractPresent(Windows.Foundation.UniversalApiContract,7)"
```
Comment (karenbtlai) [inline dev/Expander/Expander.xaml:5]: nit: this should be moved under `xmlns:local` for consistent formatting

File: dev/Expander/Expander_themeresources.xaml [hunk 8]
```diff
@@ -2,31 +2,351 @@
 <ResourceDictionary
     xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
     xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+    xmlns:contract7Present="http://schemas.microsoft.com/winfx/2006/xaml/presentation?IsApiContractPresent(Windows.Foundation.UniversalApiContract,7)"
```
Comment (karenbtlai) [inline dev/Expander/Expander_themeresources.xaml:5]: nit: this too should be moved under xmlns:local for consistent formatting

**Review by karenbtlai — APPROVED** [review]

**Comment by mdtauk** [comment]
Why is the expander using so much padding top and bottom?

image

and not closer in line with the ComboBox and DropDownButton controls?

image
